dev/com.ibm.ws.security.common.jsonwebkey/src/com/ibm/ws/security/common/jwk/impl/JwKRetriever.java

```diff
@@ -466,6 +473,45 @@ protected Key doJwkRemote(String kid, String x5t, String use, boolean useSystemP
 
         return getJwkFromJWKSet(locationUsed, kid, x5t, use, jsonString, keyType);
     }
+    
+    private String logCWWKS6049E(String url, int iStatusCode, String errMsg) {
+        // TODO - Message will be added to .nlsprops file under 222394
+        String defaultMessage = "CWWKS6049E: A JSON Web Key (JWK) was not returned from the URL [" + url
+                + "]. The response status was [" + iStatusCode + "] and the content returned was [" + errMsg + "].";
+
+        String message = TraceNLS.getFormattedMessage(getClass(),
+                "com.ibm.ws.security.jwt.internal.resources.JWTMessages", "JWT_JWK_RETRIEVE_FAILED",
+                new Object[] { url, Integer.valueOf(iStatusCode), errMsg }, defaultMessage);
+        Tr.error(JwKRetriever.tc, message, new Object[0]);
+        return message;
+    }
+    
+    @FFDCIgnore({ SocialLoginWrapperException.class })
+    protected String getHTTPRequestAsString(HttpClient client, String url) throws Exception, IOException {
+        String jsonString = null;
+        
+        try {
+            jsonString = httpUtils.getHttpJsonRequest(client, url);
+        } catch (SocialLoginWrapperException sle) {
+            Throwable cause = sle.getCause();
+            if (cause instanceof HttpResponseNullOrEmptyException) {
+                HttpResponseNullOrEmptyException ex = (HttpResponseNullOrEmptyException) cause;
+                throw new Exception(logCWWKS6049E(url, ex.getStatusCode(), jsonString));
+            } else if (cause instanceof HttpResponseNot200Exception) {
+                HttpResponseNot200Exception ex = (HttpResponseNot200Exception) cause;
+                throw new Exception(logCWWKS6049E(url, ex.getStatusCode(), jsonString));
+            } else if (cause instanceof IOException) {
+                IOException ioex = (IOException) cause;
+                logCWWKS6049E(url, 0, "IOException: " + ioex.getMessage() + " " + ioex.getCause());
+                throw ioex;
+            }
+        } catch (IOException ioex) {
+            logCWWKS6049E(url, 0, "IOException: " + ioex.getMessage() + " " + ioex.getCause());
+            throw ioex;
+        }
```

Comment:
```suggestion
        } catch (SocialLoginWrapperException sle) {
            throw new Exception(logCWWKS6049E(url, sle.getStatusCode(), sle.getNlsMessage()));
        } catch (IOException ioex) {
            logCWWKS6049E(url, 0, "IOException: " + ioex.getMessage() + " " + ioex.getCause());
            throw ioex;
        }
```
The `SocialLoginWrapperException` object has all the information you need to emit the appropriate NLS message, so this block should be able to be simplified.

Reply:
I agree. That's a simpler approach.